test/filter.test.js

```diff
@@ -0,0 +1,42 @@
+import filter from '../src/filter';
+
+const active = ({ active }) => active;
+
+const users = [
+  { user: 'barney', active: true },
+  { user: 'fred', active: false }
+];
+const activeUsers = [{ user: 'barney', active: true }];
+
+describe('filter', () => {
+  describe('filters active users', () => {
+    it('one user is active', () => {
+      const result = filter(users.filter(active), active);
+      expect(result).toIncludeAllMembers(activeUsers);
+    });
+    it('some users are active', () => {
+      const result = filter(users, active);
+      expect(result).toIncludeAllMembers(activeUsers);
+    });
+  });
+  describe('returns an empty array if', () => {
+    it('no users are active', () => {
+      const result = filter(
+        users.map(({ user }) => ({ user, active: false })),
+        active
+      );
+      /* Expecting an empty array, since no users are active */
+      expect(result).toBeArrayOfSize(0);
+    });
+    it('there is no users', () => {
+      const result = filter([], active);
+      /* Expecting an empty array, since there is no users */
+      expect(result).toBeArrayOfSize(0);
+    });
+    it('users array is null', () => {
+      const result = filter(null, active);
+      /* Expecting an empty array, since array is null */
+      expect(result).toBeArrayOfSize(0);
```

Comment:
These fail, returning `[[]]` instead of `[]`. I'd expect filter to return an empty array if none of the values match the predicate. Do you agree @Liljenmaa?

Reply:
Yes